pynsxv/library/libutils.py

```diff
@@ -95,12 +95,13 @@ def get_scope(client_session, transport_zone_name):
     try:
         vdn_scopes = client_session.read('vdnScopes', 'read')['body']
         vdn_scope_list = client_session.normalize_list_return(vdn_scopes['vdnScopes'])
-        vdn_scope = [scope['vdnScope'] for scope in vdn_scope_list
-                     if scope['vdnScope']['name'] == transport_zone_name][0]
+	vdn_scope = [scope['vdnScope'] for scope in vdn_scope_list
+			if scope['vdnScope'][0]['name'] == transport_zone_name][0]
```

Comment:
I was err ing out with indice ref must be int not str. On lswitch create


Sent from Samsung Mobile


-------- Original message --------
From: Yves Fauser
Date:08-11-2016 4:25 AM (GMT-05:00)
To: vmware/pynsxv
Cc: Gavin Barnard ,Author
Subject: Re: [vmware/pynsxv] add test for empty cluster in nsx_usage.py (#51)


In pynsxv/library/libutils.py<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_vmware_pynsxv_pull_51-23discussion-5Fr74383810&d=CwMFaQ&c=Sqcl0Ez6M0X8aeM67LKIiDJAXVeAw-YihVMNtXt-uEs&r=6h6WRoU9RQXcy0gmjS8cmWw41bWf73j9c4rn9HlVRDU&m=KUNa2-tsiiLySx3vy4FMOhLm8_JzEsFAdPTmPRWvq_8&s=iRF8k7kmj0Ki6Y8LycTjajfdoTIpcNhlaSm1Ca2vRW8&e=>:

> @@ -95,12 +95,13 @@ def get_scope(client_session, transport_zone_name):
>      try:
>          vdn_scopes = client_session.read('vdnScopes', 'read')['body']
>          vdn_scope_list = client_session.normalize_list_return(vdn_scopes['vdnScopes'])
> -        vdn_scope = [scope['vdnScope'] for scope in vdn_scope_list
> -                     if scope['vdnScope']['name'] == transport_zone_name][0]
> +     vdn_scope = [scope['vdnScope'] for scope in vdn_scope_list
> +                     if scope['vdnScope'][0]['name'] == transport_zone_name][0]


scope['vdnScope'] is a dict, so scope['vdnScope'][0] will cause a KeyError. This will lead to the function returning None, None all the time, as the try/except block triggers on KeyErrors. With this change the logical switch creation will never work. What was your intend with this change?

-
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_vmware_pynsxv_pull_51_files_9db0e23aba874628e9301c4fb41e4f87f3a0b367-23r74383810&d=CwMFaQ&c=Sqcl0Ez6M0X8aeM67LKIiDJAXVeAw-YihVMNtXt-uEs&r=6h6WRoU9RQXcy0gmjS8cmWw41bWf73j9c4rn9HlVRDU&m=KUNa2-tsiiLySx3vy4FMOhLm8_JzEsFAdPTmPRWvq_8&s=-XCHB9LRiHbFTPoEoC2n_8bDhDqB64Y4cDkoVDd10jQ&e=>, or mute the thread<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_AT-5FV5r5oiEyrtcTzjwz3zE1C8gyneTAPks5qetx8gaJpZM4Jhv-2DV&d=CwMFaQ&c=Sqcl0Ez6M0X8aeM67LKIiDJAXVeAw-YihVMNtXt-uEs&r=6h6WRoU9RQXcy0gmjS8cmWw41bWf73j9c4rn9HlVRDU&m=KUNa2-tsiiLySx3vy4FMOhLm8_JzEsFAdPTmPRWvq_8&s=3HQsFfv2QtnmP8kjOew9NvUHCk67Osu1CHP_MFviPVY&e=>.